CRM/PCP/BAO/PCP.php

```diff
@@ -931,5 +931,286 @@ public static function getOwnerNotificationId($component_id, $component = 'contr
       return $ownerNotificationId;
     }
   }
+	
+	/**
+   * wrapper for ajax pcp selector.
+   *
+   * @param array $params
+   *   Associated array for params record id.
+   *
+   * @return array
+   *   associated array of pcp list
+   *   -rp = rowcount
+   *   -page= offset
+   * @todo there seems little reason for the small number of functions that call this to pass in
+   * params that then need to be translated in this function since they are coding them when calling
+   */
+  static public function getPcpListSelector(&$params) {
+    // format the params
+    $params['offset'] = ($params['page'] - 1) * $params['rp'];
+    $params['rowCount'] = $params['rp'];
+    $params['sort'] = CRM_Utils_Array::value('sortBy', $params);
+
+    // get pcps
+    $pcps = CRM_PCP_BAO_PCP::getPcpList($params);
+    $params['total'] = CRM_PCP_BAO_PCP::getPcpCount($params);
+
+    // format params and add links
+    $pcpList = array();
+    if (!empty($pcps)) {
+      foreach ($pcps as $id => $value) {
+        $value['class'][] = 'crm-entity';
+        $pcpList[$id]['class'] = $value['id'] . ',' . implode(' ', $value['class']);
+        $pcpList[$id]['pcp_id'] = $value['id'];
+        $pcpList[$id]['pcp_name'] = $value['pcp_title'];
+        $value['class'][] = 'crm-entity';
+        $pcpList[$id]['class'] = $value['id'] . ',' . implode(' ', $value['class']);
+        $pcpList[$id]['pcp_supporter'] = CRM_Utils_Array::value('pcp_supporter', $value);
+        $pcpList[$id]['pcp_page'] = CRM_Utils_Array::value('pcp_page', $value);
+        $pcpList[$id]['pcp_start_date'] = CRM_Utils_Array::value('pcp_start_date', $value);
+        $pcpList[$id]['pcp_end_date'] = CRM_Utils_Array::value('pcp_end_date', $value) ? $value['pcp_end_date'] : '(ongoing)';
+        $pcpList[$id]['pcp_status'] = CRM_Utils_Array::value('pcp_status', $value);
+        $pcpList[$id]['action'] = CRM_Utils_Array::value('action', $value);
+      }
+      return $pcpList;
+    }
+  }
+  
+    /**
+   * This function to get list of pcps.
+   *
+   * @param array $params
+   *   Associated array for params.
+   *
+   * @return array
+   */
+  public static function getPcpList(&$params) {
+    $config = CRM_Core_Config::singleton();
```

Comment:
This line should go. It's not used. The codebase does seem to be littered with instantiation of the Config object so there may have been a historical reason but there should not be one now